### ISSUE TITLE
dotnet format produces false positives on CS0234 and CS0103

### ISSUE THREAD

**Issue by stefanloerwald** [dotnet/sdk#20126]
I have a solution with multiple projects targeting net5.0. However, I'm using net6 preview7 as my SDK.

With the new dotnet format version as part of that, I get tons of warnings for CS0234, and CS0103, despite the solution building just fine. I managed to reduce it down to the following structure:

# structure

- A
  - A.csproj
  - Class.cs
- B
  - B.csproj
  - Class.cs
  - Component.razor
- sln.sln

# file contents

- `A/A.csproj`

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <AssemblyName>Foo</AssemblyName>
    <RootNamespace>Foo</RootNamespace>
  </PropertyGroup>
  <ItemGroup>
    <ProjectReference Include="..\B\B.csproj" />
  </ItemGroup>
</Project>
```

- `A/Class.cs`

```cs
namespace A
{
    public class Bar
    {
        public void Baz()
        {
            _ = typeof(System.Func<global::B.Component>);
        }
    }
}
```

- `B/B.csproj`

```xml
<Project Sdk="Microsoft.NET.Sdk.Razor">
  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <PackageId>B</PackageId>
    <RootNamespace>B</RootNamespace>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Components" Version="5.0.9" />
    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="5.0.9" />
  </ItemGroup>
</Project>
```

- `B/Class.cs`

```cs
namespace B
{
    public static class Static
    {
    }
}
```

- `B/Component.razor`

```xml
```

(empty)

- `sln.sln`

```
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio Version 16
VisualStudioVersion = 16.0.30114.105
MinimumVisualStudioVersion = 10.0.40219.1
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "A", "A\A.csproj", "{DBF55725-D133-4A5D-8A8A-5133D95723EC}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "B", "B\B.csproj", "{6BD5E7B4-07E3-43A7-B8AD-002CA9B12B18}"
EndProject
Global
        GlobalSection(SolutionConfigurationPlatforms) = preSolution
                Debug|Any CPU = Debug|Any CPU
                Release|Any CPU = Release|Any CPU
        EndGlobalSection
        GlobalSection(SolutionProperties) = preSolution
                HideSolutionNode = FALSE
        EndGlobalSection
        GlobalSection(ProjectConfigurationPlatforms) = postSolution
                {DBF55725-D133-4A5D-8A8A-5133D95723EC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
                {DBF55725-D133-4A5D-8A8A-5133D95723EC}.Debug|Any CPU.Build.0 = Debug|Any CPU
                {DBF55725-D133-4A5D-8A8A-5133D95723EC}.Release|Any CPU.ActiveCfg = Release|Any CPU
                {DBF55725-D133-4A5D-8A8A-5133D95723EC}.Release|Any CPU.Build.0 = Release|Any CPU
                {6BD5E7B4-07E3-43A7-B8AD-002CA9B12B18}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
                {6BD5E7B4-07E3-43A7-B8AD-002CA9B12B18}.Debug|Any CPU.Build.0 = Debug|Any CPU
                {6BD5E7B4-07E3-43A7-B8AD-002CA9B12B18}.Release|Any CPU.ActiveCfg = Release|Any CPU
                {6BD5E7B4-07E3-43A7-B8AD-002CA9B12B18}.Release|Any CPU.Build.0 = Release|Any CPU
        EndGlobalSection
EndGlobal
```

# Observations

```pwsh
> dotnet --list-sdks
2.1.818 [C:\Program Files\dotnet\sdk]
3.1.412 [C:\Program Files\dotnet\sdk]
5.0.400 [C:\Program Files\dotnet\sdk]
6.0.100-preview.7.21379.14 [C:\Program Files\dotnet\sdk]
```

```pwsh
> dotnet build
Microsoft (R) Build Engine version 17.0.0-preview-21378-03+d592862ed for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  B -> C:\path\to\sln\B\bin\Debug\net5.0\B.dll
  A -> C:\path\to\sln\A\bin\Debug\net5.0\A.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.34
```

```pwsh
> dotnet format -v d
  The dotnet format version is '6.0.237103+afcd94017e2877458d2e07a8844130442cf20241'.
  The dotnet runtime version is '6.0.0-preview.7.21377.19'.
  Formatting code files in workspace 'C:\path\to\sln\sln.sln'.
    Determining projects to restore...
  All projects are up-to-date for restore.
  Project A is using configuration from 'C:\path\to\sln\A\obj\Debug\net5.0\A.GeneratedMSBuildEditorConfig.editorconfig'.
  Project B is using configuration from 'C:\path\to\sln\B\obj\Debug\net5.0\B.GeneratedMSBuildEditorConfig.editorconfig'.
  Running 2 analyzers on A.
  Running 2 analyzers on B.
C:\path\to\sln\A\Class.cs(8,46): error CS0234: The type or namespace name 'Component' does not exist in the namespace 'B' (are you missing an assembly reference?) [C:\path\to\sln\A\A.csproj]
Unable to fix CS0234. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix CS0234. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
  Running 1 analyzers on A.
  Running 1 analyzers on B.
  Formatted code file 'C:\path\to\sln\A\Class.cs'.
  Formatted 1 of 6 files.
  Format complete in 8275ms.
```

The problem disappears if I delete `B/Class.cs`, even though the file is unrelated to the issue reported in `A/Class.cs`.

```pwsh
> git rm B/Class.cs; dotnet build; dotnet format; git restore --staged B/Class.cs; git restore B/Class.cs
Microsoft (R) Build Engine version 17.0.0-preview-21378-03+d592862ed for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  B -> C:\path\to\sln\B\bin\Debug\net5.0\B.dll
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  A -> C:\path\to\sln\A\bin\Debug\net5.0\A.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.06
  Formatting code files in workspace 'C:\path\to\sln\sln.sln'.
  Format complete in 8592ms.
```

In my actual solution, I get hundreds of these warnings, which is very annoying, because they are all wrong.

**Comment by stefanloerwald** [comment]
For your convenience, here's a repository for the issue: https://github.com/stefanloerwald/dotnet-format-issue-20126

**Comment by stefanloerwald** [comment]
The issue persists with RC1.

**Comment by stefanloerwald** [comment]
Fixed in stable (SDK 6.0.100).
